models.py

```diff
@@ -0,0 +1,153 @@
+from datetime import datetime
+from werkzeug.security import generate_password_hash, check_password_hash
+from app import db, login
+from flask_login import LoginManager, UserMixin, login_required
+
+
+# klasa z kontem użytkownika
+class User(UserMixin, db.Model):
+    id = db.Column(db.Integer, primary_key=True)
+    username = db.Column(db.String(64), index=True, unique=True)
+    email = db.Column(db.String(120), unique=True, index=True)
+    password_hash = db.Column(db.String(128))
+    profiles = db.relationship('Profile', backref='User', lazy='dynamic',
+                               cascade="all, delete, delete-orphan")  # powiązanie z kontem uzytkownika
+    reports = db.relationship('ReportPlayer', backref='Report', lazy='dynamic',
+                              cascade="all, delete, delete-orphan")
+    invitation_out = db.relationship('ClashInvitation', backref='teamleader', lazy='dynamic',
+                                     cascade="all, delete, delete-orphan")
+    request_in = db.relationship('ClashRequest', backref='team_creator', lazy='dynamic',
+                                 cascade="all, delete, delete-orphan")
+
+    def set_password(self, password):
+        self.password_hash = generate_password_hash(password)
+
+    def check_password(self, password):
+        return check_password_hash(self.password_hash, password)
+
+    def __repr__(self):
+        return '<User {}>'.format(self.username)
+
+
+# klasa z profilem, przypisywanym do konta użytkownika, w relacji jeden do wielu (1 uzytkownik wiele profili)
+class Profile(UserMixin, db.Model):
+    id = db.Column(db.Integer, primary_key=True)
+    nickname = db.Column(db.String(64), index=True, unique=True)
+    server = db.Column(db.String(10), index=True)
+    division = db.Column(db.String(20), index=True)
+    best_position = db.Column(db.String(20), index=True)
+    alternative_position = db.Column(db.String(20), index=True)
+    user_id = db.Column(db.Integer, db.ForeignKey('user.id'))
+    clash_team = db.relationship('ClashTeam', backref='host', lazy='dynamic')
+    invitation_in = db.relationship('ClashInvitation', backref='invitedguest', lazy='dynamic',
+                                    cascade="all, delete, delete-orphan")
+    request_out = db.relationship('ClashRequest', backref='candidate', lazy='dynamic',
+                                  cascade="all, delete, delete-orphan")
+    reported = db.relationship('ReportPlayer', backref='Reported', lazy='dynamic',
+                               cascade="all, delete, delete-orphan")
+
+
+# Klasa przedstawiająca tworzenie dużyny Clash
+class ClashTeam(UserMixin, db.Model):
+    id = db.Column(db.Integer, primary_key=True)
+    clash_date = db.Column(db.DateTime, nullable=False, default=None)
+    division = db.Column(db.String(64), index=True, default=None)
+    toplane = db.Column(db.String(64), index=True, default=None)
```

Comment:
If I understand this correctly then `toplane` is a position that a player can play? If yes, it seems that you want to assign a player to the position. Do not use nickname but always use the user ID this should be as a foreign key.

I may be wrong because I don't understand the game. Bartosz will be able to explain the database normalisation: https://en.wikipedia.org/wiki/Database_normalization